src/data/convai2_dataset.py

```diff
@@ -79,26 +104,29 @@ def collate_fn(self, samples: list[ConvAI2Dialog], return_all_candidates: bool =
         return_all_candidates = self.have_candidates & return_all_candidates
         str_contexts = [" ".join(sample.context) for sample in samples]
         # [batch size, context seq len]
-        b_contexts = self.context_tokenizer(
-            str_contexts,
-            max_length=self.max_context_len,
-            padding=True,
-            truncation=True,
-            return_tensors="pt",
-            add_special_tokens=False,
-        ).input_ids
+        b_contexts = self.context_tokenizer(str_contexts, max_length=self.max_context_len, **self.tokenizer_kwargs)
+
+        str_conditions = []
+        if self.condition is Conditions.YOUR:
+            str_conditions = [" ".join(sample.my_persona) for sample in samples]  # type: ignore
```

Comment:
Why are you ignoring the mypy here? What error does it raise?

Reply:
It raises `Argument 1 to "join" of "str" has incompatible type "Optional[List[str]]"; expected "Iterable[str]"  [arg-type]` error. Although the condition guarantees the existence of the `my_persona` attribute.